src/ddscxx/tests/CMakeLists.txt

```diff
@@ -83,7 +94,13 @@ if(DEFINED ENV{SYSTEM_TEAMFOUNDATIONSERVERURI})
   endif()
 endif()
 
-set_property(TARGET ddscxx_tests PROPERTY CXX_STANDARD 17)
+if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  # Microsoft does not specify the __cplusplus macro by default
+  # https://docs.microsoft.com/en-us/cpp/build/reference/zc-cplusplus
```

Comment:
`entity_properties.hpp` gets included by application code, so doesn't this lovely "feature" of MSVC++ now break C++ application code?

I suppose we could force `/Zc:__cplusplus` into the CMake config files for cyclonedds-cxx, but then simply including Cyclone would likely change the definition for the application as well.

Altogether I think it would make more sense to make sure `features.hpp.in` adds an entry for `ENABLE_LEGACY` (obviously it would have to be a slightly different name, like `DDSCXX_USE_BOOST` or so), and then use that instead of checking `__cplusplus`.